staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -5088,6 +5088,27 @@ type PodStatus struct {
 	// +featureGate=DynamicResourceAllocation
 	// +optional
 	ResourceClaimStatuses []PodResourceClaimStatus `json:"resourceClaimStatuses,omitempty" patchStrategy:"merge,retainKeys" patchMergeKey:"name" protobuf:"bytes,15,rep,name=resourceClaimStatuses"`
+
+	// AllocatedResources is the total requests allocated for this pod by the node.
+	// Kubelet sets this to the accepted requests when a pod (or resize) is admitted.
+	// If pod-level requests are not set, this will be the total requests aggregated
+	// across containers in the pod.
+	// +featureGate=InPlacePodVerticalScaling
+	// +featureGate=PodLevelResources
+	// +optional
+	AllocatedResources ResourceList `json:"allocatedResources,omitempty" protobuf:"bytes,18,rep,name=allocatedResources,casttype=ResourceList,castkey=ResourceName"`
+
+	// Resources represents the compute resource requests and limits that have been
+	// applied at the pod level. If pod-level resources are not explicitly specified,
+	// then these will be the aggregate resources computed from containers. If limits are
+	// not defined for all containers (and pod-level limits are also not set), those
+	// containers remain unrestricted, and no aggregate pod-level limits will be applied.
+	// Pod-level limit aggregation is only performed, and is meaningful only, when all
+	// containers have defined limits.
+	// +featureGate=InPlacePodVerticalScaling
+	// +featureGate=PodLevelResources
+	// +optional
+	Resources *ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,19,opt,name=resources"`
```

Comment:
(for API discussion) it was maybe a mistake that we reused this type for container resources, since the `Claims` field is unused. Should we create a new type to use here, or stick with ResourceRequirements for consistency with the container level fields?